commons/src/main/java/org/infinispan/commons/util/EmptyQueue.java

```diff
@@ -0,0 +1,104 @@
+package org.infinispan.commons.util;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Queue;
+
+import static java.util.Collections.emptyIterator;
+
+/**
+ * @author gustavonalle
+ * @since 7.2
+ */
+public final class EmptyQueue<T> implements Queue<T> {
+
+   @Override
+   public int size() {
+      return 0;
+   }
+
+   @Override
+   public boolean isEmpty() {
+      return true;
+   }
+
+   @Override
+   public boolean contains(Object o) {
+      return false;
+   }
+
+   @Override
+   public Iterator<T> iterator() {
+      return emptyIterator();
+   }
+
+   @Override
+   public Object[] toArray() {
+      return new Object[0];
+   }
+
+   @Override
+   public <T1> T1[] toArray(T1[] a) {
+      throw new UnsupportedOperationException();
```

Comment:
The ```EmptyQueue``` is just a workaround to conditionally remove an entry from the map when the value is empty:
```map.remove(key, emptyQueue)```  that's why most of its methods are useless  :smile:
```